tests/test-generators.py

```diff
@@ -26,56 +26,80 @@ def parse_output(result, error_code) -> str:
 
 for g in sorted(glob.glob('generators/*.clvm')):
     print(f"{g}")
-    start_time = time()
-    error_code, result = run_gen(g)
-    run_time = time() - start_time
+    sys.stdout.write("running generator...\r")
+    start_time = perf_counter()
+    error_code, result = run_gen(g, 0, "generators/block-834752.hex")
+    run_time = perf_counter() - start_time
     output = parse_output(result, error_code)
 
-    start_time = time()
-    error_code2, result2 = run_gen(g, MEMPOOL_MODE)
-    run_time2 = time() - start_time
+    sys.stdout.write("running generator (mempool mode) ...\r")
+    sys.stdout.flush()
+    start_time = perf_counter()
+    error_code2, result2 = run_gen(g, MEMPOOL_MODE, "generators/block-834752.hex")
+    run_time2 = perf_counter() - start_time
     output2 = parse_output(result2, error_code2)
 
     with open(g) as f:
         expected = f.read().split('\n', 1)[1]
         if not "STRICT" in expected:
             expected2 = expected
-            if not (result is None and result2 is None or result.cost == result2.cost):
+            if not (result is None or result2 is None or result.cost == result2.cost):
                 print("cost when running in mempool mode differs from normal mode!")
                 failed = 1
         else:
             expected, expected2 = expected.split("STRICT:\n", 1)
 
-        compare_output(output, expected, "")
-        print(f"  run-time: {run_time:.2f}s")
+        sys.stdout.write("\x1b[K")
+        sys.stdout.flush()
+
+        if run_time != 0:
+            compare_output(output, expected, "")
+            print(f"  run-time: {run_time:.2f}s")
 
         compare_output(output2, expected2, "STRICT")
         print(f"  run-time: {run_time2:.2f}s")
 
+        # this is the ambition with future optimizations
         limit = 1.5
+        strict_limit = 1.5
 
         # temporary higher limits until this is optimized
         if "duplicate-coin-announce.clvm" in g:
```

Comment:
long(er) run, i might do:

    special_limits_per_test = {
        "duplicate-coin-announce.clvm": {"limit": 10, "strict_limit": 5},
        # ...
    }
    default_limits = { "limit": 1.5, "strict_limit": 1.5 }

    # ...

    filename = os.path.basename(g)
    use_limits = special_limits_per_test.get(filename, default_limits)
    limit, strict_limit = use_limits["limit"], use_limits["strict_limit"]

And since they're dict you can add more breadth.

Reply:
yeah, that's fair. I've been hoping that these exceptions are short lived (hoping to optimize `clvm` to be faster). It's not so short lived by now though